apps/api/e2e/idempotency.e2e.ts

```diff
@@ -0,0 +1,282 @@
+import { UserSession } from '@novu/testing';
+import { CacheService } from '@novu/application-generic';
+import { expect } from 'chai';
+describe('Idempotency Test', async () => {
+  let session: UserSession;
+  const path = '/v1/testing/idempotency';
+  const HEADER_KEYS = {
+    IDEMPOTENCY: 'idempotency-key',
+    RETRY_AFTER: 'retry-after',
+    IDEMPOTENCY_CONFLICT: 'x-idempotency-conflict',
+  };
+  let cacheService: CacheService | null = null;
+
+  describe('when enabled', () => {
+    before(async () => {
+      session = new UserSession();
+      await session.initialize();
+      cacheService = session.testServer?.getService(CacheService);
+      process.env.API_IDEMPOTENCY_ENABLED = 'true';
+    });
+
+    it('should return cached same response for duplicate requests', async () => {
+      const key = Math.random().toString();
```

Comment:
Tests should be deterministic, and `Math.random` is not. Without determinism, tests can be flakey.

Can we replace all use of `Math.random` in this file with static values?